e2e/playwright/actions/src/tests/share/share-file.spec.ts

```diff
@@ -0,0 +1,246 @@
+      });
+
+      test('[C286330] Copy shared file URL', async ({ personalFiles, page }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file4, 'Share'));
+
+        const url = await personalFiles.shareDialog.getLinkUrl();
+        expect(url).toContain(shareLinkPreUrl);
+
+        await personalFiles.shareDialog.urlAction.click();
+
+        const shareSnackBar = personalFiles.snackBar;
+        await expect(shareSnackBar.getByMessageLocator('Link copied to the clipboard')).toBeVisible();
+
+        await page.goto(url);
+        await personalFiles.viewer.waitForViewerToOpen();
+
+        const downloadPromise = personalFiles.page.waitForEvent('download');
+        await personalFiles.viewer.toolbar.sharedDownloadButton.click();
+
+        const download = await downloadPromise;
+        expect(download.suggestedFilename()).toBe(file4);
+      });
+
+      test('[C286332] Share a file with expiration date', async ({ personalFiles, nodesApiAction, page }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file5, 'Share'));
+
+        await personalFiles.shareDialog.expireToggle.click();
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(true);
+
+        await personalFiles.shareDialog.datetimePickerButton.click();
+        expect(await personalFiles.shareDialog.dateTimePicker.isCalendarOpen()).toBe(true);
+
+        await personalFiles.shareDialog.dateTimePicker.pickDateTime();
+
+        const inputDate = await personalFiles.shareDialog.getExpireDate();
+
+        await page.waitForTimeout(timeouts.normal);
+        const expireDateProperty = await nodesApiAction.getSharedExpiryDate(file5Id);
+
+        expect(Utils.formatDate(expireDateProperty)).toEqual(Utils.formatDate(inputDate));
+      });
+
+      test('[C286337] Expire date is displayed correctly', async ({ personalFiles, nodesApiAction }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file6, 'Share'));
+        const expireProperty = await nodesApiAction.getSharedExpiryDate(file6Id);
+
+        expect(expireProperty).toEqual(expiryDate);
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(true);
+        expect(Utils.formatDate(await personalFiles.shareDialog.getExpireDate())).toEqual(Utils.formatDate(expiryDate));
+      });
+
+      test('[C286333] Disable the share link expiration', async ({ personalFiles, nodesApiAction, page }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file7, 'Share'));
+
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(true);
+        expect(await personalFiles.shareDialog.getExpireDate()).not.toBe('');
+
+        await personalFiles.shareDialog.expireToggle.click();
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(false);
+
+        page.waitForTimeout(timeouts.tiny);
```

Comment:
await is missing,  Also we can have page from  personalFiles.page

Reply:
added await. waitForTimeout is an inbuilt method of Page Interface provided by Playwright so I guess we don't need  personalFiles.page